packages/neon_framework/lib/src/storage/request_cache.dart

```diff
@@ -1,31 +1,34 @@
 import 'dart:async';
+import 'dart:convert';
 import 'dart:typed_data';
 
 import 'package:logging/logging.dart';
 import 'package:meta/meta.dart';
 import 'package:neon_framework/models.dart';
 import 'package:neon_framework/src/platform/platform.dart';
-import 'package:neon_framework/src/utils/request_manager.dart';
-import 'package:nextcloud/utils.dart';
 import 'package:path/path.dart' as p;
 import 'package:path_provider/path_provider.dart';
 import 'package:sqflite_common_ffi/sqflite_ffi.dart';
-import 'package:timezone/timezone.dart' as tz;
 
 final _log = Logger('RequestCache');
 
-/// A storage used to cache a key value pair.
+/// A storage used to cache HTTP request status codes, bodies and headers.
 abstract interface class RequestCache {
-  /// Get's the cached value and parameters for the given [key].
-  Future<({Uint8List value, CacheParameters? parameters})?> get(Account account, String key);
-
-  /// Set's the cached [value] at the given [key].
+  /// Gets the cached status code, body and headers for the given [key].
+  Future<
+      ({
+        int statusCode,
+        Uint8List body,
+        Map<String, String> headers,
+      })?> get(Account account, String key);
+
+  /// Set's the cached [body] at the given [key].
   ///
-  /// If a value is already present it will be updated with the new one.
-  Future<void> set(Account account, String key, Uint8List value, CacheParameters? parameters);
+  /// If a body is already present it will be updated with the new one.
+  Future<void> set(Account account, String key, int statusCode, Uint8List body, Map<String, String> headers);
```

Comment:
What about passing a http.BaseResponse parameter directly?

Reply:
It needs to be http.Request, but yeah I didn't think about that.